app/jobs/regular/export_user_archive.rb

```diff
@@ -118,6 +118,14 @@ class ExportUserArchive < ::Jobs::Base
 
     def execute(args)
       @current_user = User.find_by(id: args[:user_id])
+
+      @requesting_user = User.find_by(id: args[:requesting_user_id]) || @current_user
+      if !(@requesting_user&.admin? || @requesting_user == @current_user)
```

Comment:
In this case why does it matter if the requesting user is the same as the current user?